src/lib/searchUtilActions/search.ts

```diff
@@ -86,6 +89,58 @@ export async function handleAasDiscoverySearch(searchAssetId: string): Promise<s
     }
 }
 
+export type AasData = {
+    submodelDescriptors: SubmodelDescriptor[] | undefined;
+    aasRegistryRepositoryOrigin: string | undefined;
+};
+
+export type AasSearchResult = {
+    aasUrl: string;
+    aas: AssetAdministrationShell | null;
+    aasData: AasData | null;
+};
+
+export async function handleSearchForAas(val: string): Promise<AasSearchResult> {
+    const repositoryClient = new AssetAdministrationShellRepositoryApi({
+        basePath: process.env.AAS_REPO_API_URL,
+        fetch: mnestixFetch(),
+    });
```

Comment:
the repositoryClient  is already defined in the 'ApiProvider.tsx' but I cannot use it here, as this is not a react component and the ApiProvider works with a context. I have seen this type of code duplication elsewhere, so I think it is not too bad.
Ideas are highly welcome.